include/sys/__assert.h

```diff
@@ -63,6 +63,10 @@
 
 #include <stdbool.h>
 
+#ifdef __cplusplus
```

Comment:
This has to be moved into the `#if __ASSERT_ON` block and be positioned after `#include <sys/printk.h>`.